fms/modules/embedding.py

```diff
@@ -238,3 +236,65 @@ def forward(self, inp, reverse=False):
         rank = torch.tensor(self.rank)
         world_size = torch.tensor(self.world_size)
         return all_gather_from_tensor_model_parallel_region(out_par, rank, world_size)
+
+
+class TPEmbedding(nn.Embedding, TPModule):
```

Comment:
would help to clarify the differences between TPEmbedding vs TPWordEmbedding

Reply:
will add some more documentation